dev/plugins/hu.elte.txtuml.export.javascript/src/hu/elte/txtuml/export/javascript/Exporter.java

```diff
@@ -0,0 +1,178 @@
+	 */
+	private void prepare() throws IOException {
+		ResourceHandler.copyResourcesTo(target);
+	}
+
+	/**
+	 * Writes the JSON serialized model in a JavaScript wrapper to the target
+	 * folder as input.js
+	 * 
+	 * @throws IOException
+	 * @throws JAXBException
+	 */
+	private void insertInput() throws IOException, JAXBException {
+
+		try (FileWriter fw = new FileWriter(Paths.get(target, "input.js").toFile());
+				BufferedWriter jsfile = new BufferedWriter(fw);) {
+			jsfile.write("var input = ");
+			JSONExporter.writeObjectAsJSON(model, jsfile);
+			jsfile.write(";");
+
+		} catch (IOException e) {
+			throw e;
+		} catch (JAXBException e) {
+			throw e;
+		}
+	}
+
+	/**
+	 * Opens visualize.html in the default browser
+	 * 
+	 * @throws PartInitException
+	 * @throws MalformedURLException
+	 */
+	private void display() throws PartInitException, MalformedURLException {
+		// set to use external browser due to internal browser bug
+		// https://bugs.eclipse.org/bugs/show_bug.cgi?id=501978
+		final IWebBrowser browser = PlatformUI.getWorkbench().getBrowserSupport().getExternalBrowser();
+		browser.openURL(Paths.get(target, "visualize.html").toUri().toURL());
+	}
+
+	/**
+	 * Collects model and layout information of the diagrams the Exporter is
+	 * initialized with, then generates the necessary files for JointJS
+	 * visualization, finally it opens the visualize.html in the default browser
+	 * which will visualize the diagrams
+	 * 
+	 * @throws ModelMapException
+	 * @throws ArrangeException
+	 * @throws UnknownDiagramTypeException
+	 * @throws IOException
+	 * @throws JAXBException
+	 * @throws PartInitException
+	 * @throws UnexpectedEndException
+	 */
+	public void export() throws ModelMapException, ArrangeException, UnexpectedDiagramTypeException, IOException,
+			JAXBException, PartInitException, UnexpectedEndException {
+		createModel();
+		prepare();
+		insertInput();
+		display();
+	}
+
+	/**
+	 * Populates model with the diagrams to be exported
+	 * 
+	 * @throws ModelMapException
+	 * @throws ArrangeException
+	 * @throws UnknownDiagramTypeException
+	 * @throws UnexpectedEndException
+	 */
+	private void createModel()
+			throws ModelMapException, ArrangeException, UnexpectedDiagramTypeException, UnexpectedEndException {
+
+		for (Pair<String, DiagramExportationReport> report : layout.getReportsWithDiagramNames()) {
+			String name = report.getFirst();
+			DiagramExportationReport der = report.getSecond();
+
+			LayoutVisualizerManager lvm = new LayoutVisualizerManager(der.getNodes(), der.getLinks(),
+					der.getStatements());
+			lvm.arrange();
+			// lvm now contains all the diagram elements abstract position and
+			// size informations
+
+			ModelMapProvider map = new ModelMapProvider(URI.createFileURI(genFolder), der.getModelName());
+			// map connects the layout information to the EMF-UML model
+			// informations
+			List<Statement> statements = lvm.getStatementsSet();
+
+			double spacing = 0.8; // default spacing
```

Comment:
It seems you're using the spacing as some kind of scaling for the result boxes. 

The normal behavior of the arranging algorithm is that the space between the individual boxes is exactly the width and height of a box.  The user can adjust this and specify the percentage of these width and height values. The algorithm take this setting into consideration and the resulting box coordinates and dimensions will be set accordingly.

There is an interesting part in the story however. The algorithm generates the layout on a uniform grid, where the boxes are placed on the points of this grid (without any dimension). But in the link arrange phase the width and height of the boxes matter, and the grid has to be refined. To be able to determine the relative dimensions of the boxes each of them is feeded with an initial (desired) pixel value. This introduces a measure, the `pixel-grid-ratio` which defines how many pixels can fit on a grid. This can be queried from the [LayoutVisualizer](https://github.com/ELTE-Soft/txtUML/blob/master/dev/plugins/hu.elte.txtuml.layout.visualizer/src/hu/elte/txtuml/layout/visualizer/algorithms/LayoutVisualize.java#L105)

These whole process results arranged rectangles with normal (width, height) dimensions and the so called pixelDimension (pixelWidth pixelHeight). Their Position is defined on the grid though, so they may be scaled up with the `pixel-grid-ratio`. I'm doing this with the [LayoutTransformer](https://github.com/ELTE-Soft/txtUML/blob/master/dev/plugins/hu.elte.txtuml.export.papyrus/src/hu/elte/txtuml/export/papyrus/elementsarrangers/txtumllayout/LayoutTransformer.java)